src/arch/ia32/descriptor/tss/permissions.rs

```diff
@@ -0,0 +1,139 @@
+//! A module containing the implementation and tests for the [`Permissions`]
+//! structure.
+use crate::arch::ia32::PrivilegeLevel;
+use bit_field::BitField;
+use core::fmt;
+
+/// A structure representing the bits 8 to 15 from a
+/// [TssDescriptor](super::TssDescriptor) structure.
+///
+/// It gathers the following fields:
+/// - Segment type (TYPE)
+/// - Descriptor privilege level (DPL)
+/// - Present bit (P)
+#[derive(Copy, Clone)]
+pub struct Permissions(u8);
+
+/// The set of all field offsets for the [`Permissions`] structure.
+mod offset {
+    /// Offset of the busy bit (B) within the [Permissions](super::Permissions)
+    /// structure
+    pub const BUSY: usize = 1;
+
+    /// Bounds of the descriptor privilege level (DPL) bits within the
+    /// [Permissions](super::Permissions) structure.
+    pub mod privilege_level {
+        /// Lower bit offset.
+        pub const LOWER: usize = 5;
+
+        /// Upper bit offset.
+        pub const UPPER: usize = 6;
+    }
+
+    /// Offset of the present bit (P) within the
+    /// [Permissions](super::Permissions) structure.
+    pub const PRESENT: usize = 7;
+}
+
+impl Default for Permissions {
+    fn default() -> Self {
+        Permissions(0b00001001)
```

Comment:
Here again, do you want to precise what is set ?

Reply:
Done in ce93a17